src/gmmtypes.jl

```diff
@@ -64,8 +64,8 @@ types are supported: full covarariance and diagonal covariance.
     hist::Vector{History}
     "number of points used to train the GMM"
     nx::Int
-    function GMM(w::Vector{T}, μ::Matrix{T}, Σ::Union{DiagCov{T},FullCov{T}},
-                 hist::Vector, nx::Int)
```

Comment:
Ah, whoops, this was my bad.  I didn't fix this one properly.  The `where{T, CT}` should instead read `where{T, CT<:Union{DiagCov{T},FullCov{T}}`.  Of course it still works because I've just removed a type restriction.  :P

The reason why I've changed so many things to say `AbstractArray{T,2}` instead of `Matrix{T}` is because 0.6 introduced the `RowVector` type, which satisfies `RowVector{T} <: AbstractArray{T,2}`, but does not satisfy `RowVector{T} <: Matrix{T}`.  So if you want things like `ones(10)'` to be able to passed to these functions, you need to use `AbstractArray{T,2}`.

You can, of course, use `AbstractMatrix{T}` as well.